splinepy/utils/data.py

```diff
@@ -0,0 +1,224 @@
+
+        no more tracking.
+        """
+        return np.array(self, copy=True)
+
+    def view(self, *args, **kwargs):
+        """Set writeable flags to False for the view."""
+        v = super(self.__class__, self).view(*args, **kwargs)
+        v.flags.writeable = False
+        return v
+
+    def __iadd__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__iadd__(*args, **kwargs)
+
+    def __isub__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__isub__(*args, **kwargs)
+
+    def __imul__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__imul__(*args, **kwargs)
+
+    def __idiv__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__idiv__(*args, **kwargs)
+
+    def __itruediv__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__itruediv__(*args, **kwargs)
+
+    def __imatmul__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__imatmul__(*args, **kwargs)
+
+    def __ipow__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__ipow__(*args, **kwargs)
+
+    def __imod__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__imod__(*args, **kwargs)
+
+    def __ifloordiv__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__ifloordiv__(*args, **kwargs)
+
+    def __ilshift__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__ilshift__(*args, **kwargs)
+
+    def __irshift__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__irshift__(*args, **kwargs)
+
+    def __iand__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__iand__(*args, **kwargs)
+
+    def __ixor__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__ixor__(*args, **kwargs)
+
+    def __ior__(self, *args, **kwargs):
+        self._set_modified()
+        return super(self.__class__, self).__ior__(*args, **kwargs)
+
+    def __setitem__(self, *args, **kwargs):
+        self._set_modified()
+        super(self.__class__, self).__setitem__(*args, **kwargs)
+
+    def __setslice__(self, *args, **kwargs):
+        self._set_modified()
+        super(self.__class__, self).__setslice__(*args, **kwargs)
+
+    def __getslice__(self, *args, **kwargs):
+        self._set_modified()
+        """
+        return slices I am pretty sure np.ndarray does not have __*slice__
+        """
+        slices = super(self.__class__, self).__getitem__(*args, **kwargs)
+        if isinstance(slices, np.ndarray):
+            slices.flags.writeable = False
```

Comment:
```suggestion
        """
        return slices I am pretty sure np.ndarray does not have __*slice__
        """
        self._set_modified()
        slices = super(self.__class__, self).__getitem__(*args, **kwargs)
        if isinstance(slices, np.ndarray):
            slices.flags.writeable = False
```
I just wanted to make a comment. Even if you keep this function, please change the order here

Reply:
nice catch!